### PR TITLE
Logging fixes and enhancements.

### PR DESCRIPTION
**Pull request by cmccoy**
Addresses #66, #67.

### CODE REVIEW

**Comment by voellm** [comment]
LGTM

**Comment by voellm** [comment]
Connor - Please complete this today so it will be included in v0.14.0.  THanks!

**Comment by cmccoy** [comment]
test this please.

**Comment by cmccoy** [comment]
Rebased and fixed some merge conflicts - PTAL.

File: perfkitbenchmarker/publisher.py [hunk 1]
```diff
@@ -227,12 +228,16 @@ def _FormatMetadata(self, metadata):
                     for k, v in sorted(metadata.iteritems()))
 
   def PublishSamples(self, samples):
+    stream = io.BytesIO()
     dashes = '-' * 25
-    self.stream.write('\n' + dashes +
-                      'PerfKitBenchmarker Results Summary' +
-                      dashes + '\n')
+    stream.write('\n' + dashes +
```
Comment (voellm) [inline perfkitbenchmarker/publisher.py]: Why did you remove "self." here are leave it below?

File: perfkitbenchmarker/publisher.py [hunk 2]
```diff
@@ -227,12 +228,16 @@ def _FormatMetadata(self, metadata):
...
+                 'PerfKitBenchmarker Results Summary' +
+                 dashes + '\n')
 
     if not samples:
+      logging.debug('Pretty-printing results to %s:\n%s', self.stream,
+                    stream.getvalue())
+      self.stream.write(stream.getvalue())
```
Comment (voellm) [inline perfkitbenchmarker/publisher.py]: Example line where self is still present.

File: perfkitbenchmarker/publisher.py (hunk 1 above)
Comment (cmccoy) [inline perfkitbenchmarker/publisher.py]: This change is in order to print the results to two locations: `self.stream` and the log.  The local variable `stream` is used to build the full formatted text output, then that message (via `stream.getvalue()`) is written to `self.stream` and `logging.debug`.

Would a different variable name make this clearer? Perhaps `buffer`?

Comment (voellm) [inline perfkitbenchmarker/publisher.py]: Clearer and less error prone for future updates + some comments :)

On Fri, Apr 24, 2015 at 3:01 PM, Connor McCoy <notifications@github.com>
wrote:

> In perfkitbenchmarker/publisher.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/68#discussion_r29089576>
> :
>
> >      dashes = '-' * 25
> > -    self.stream.write('\n' + dashes +
> > -                      'PerfKitBenchmarker Results Summary' +
> > -                      dashes + '\n')
> > +    stream.write('\n' + dashes +
>
> This change is in order to print the results to two locations: self.stream
> and the log. The local variable stream is used to build the full
> formatted text output, then that message (via stream.getvalue()) is
> written to self.stream and logging.debug.
>
> Would a different variable name make this clearer? Perhaps buffer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/68/files#r29089576>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/

**Comment by cmccoy** [comment]
test this please.

**Comment by cmccoy** [comment]
@voellm: Thanks for looking. I changed the variable name and added a comment.

**Comment by voellm** [comment]
LGTM

**Comment by cmccoy** [comment]
test this please.

**Comment by cmccoy** [comment]
test this please.
